db/migrate/20200109223351_create_build_failures.rb

```diff
@@ -0,0 +1,12 @@
+class CreateBuildFailures < ActiveRecord::Migration
+  def change
+    create_table :build_failures do |t|
+      t.integer :repo_id
+      t.integer :branch_id
+      t.integer :travis_build_id
+
+      t.datetime :last_notified_at
+      t.timestamps
```

Comment:
Could this just be a last_notified_at column on the branch and thus not need a whole table?

Reply:
Possibly.  I definitely took the approach of "haphazardly try and get this working" for this one, so most likely.  I also didn't want to just modify a table in case we wanted to separate concerns a bit.

That said, if the `travis_build_id` isn't being used here (definitely possible, since I forget if I even call it), then yes, I think we can get rid of this.  Or at least move those two columns to `Branch` and avoid the new model.  I will look into doing that today yet.

Reply:
Moved the changes into the `Branch` model.  I did keep the `travis_build_id` (as `travis_build_failure_id`), but mostly worked just fine.

Thanks for the simplification suggestion!